From 1ea3f5ed2928bc1cff97e2856a7201135569eab5 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 09 Aug 2021 08:43:32 +0300
Subject: [PATCH] [SW-50880] ppboot: greco: arc: remove delay from halt n reset

For performing PCIe initialization on ARC0, halt and reset for
ARC0 needs to be performed before udelay init as part of early plat
init function. Upon doing experiement without using delay, ARC responds
very quickly to halt request without entering wait loop. This patch
removes udelay from halt and reset function of ARC and increase
number of retry.

Change-Id: I5d5df4550b3770ac05993b8efea11bc9f6799a7f
---

diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 7b59da6..b4c4a85 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -68,7 +68,7 @@
 #define PCIE_EXT_LD_TIMEOUT_US		576
 
 #define PCIE_WAIT_TIMEOUT_US		(5 * USEC_PER_SEC)
-#define ARC_HALT_ACK_RETRIES	10
+#define ARC_HALT_ACK_RETRIES		100
 
 
 bool hl_is_debug_mode(enum hl_debug_hw_func func)
@@ -615,7 +615,7 @@
 			val = regmap_readl(halt_ack_offset);
 			if (val & halt_ack_mask)
 				break;
-			udelay(10);
+
 			retry--;
 			if (!retry) {
 				preboot_puts_hex("No ack from ARC", id, false);
